src/main/java/core/basesyntax/ColorSupplier.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public abstract class ColorSupplier extends Figure {
+    private static final int COLOR_NUMBERS = 8;

```

Comment:
The constant `COLOR_NUMBERS` is named in a way that could be confusing. Consider renaming it to something like `COLOR_COUNT` to better reflect its purpose.